app_integrations/apps/onelogin.py

```diff
@@ -0,0 +1,192 @@
+                         'Content-Type': 'application/json'}
+
+        response = requests.post(token_url,
+                                 json={'grant_type':'client_credentials'},
+                                 headers=headers_token)
+
+        if not self._check_http_response(response):
+            return False
+
+        bearer = 'bearer:%s' % (response.json()['access_token'])
+        self._auth_headers = {'Authorization': bearer}
+
+    def _gather_logs(self):
+        """Gather the authentication log events."""
+
+        if not self._auth_headers:
+            self._auth_headers = self._generate_headers(self._ONELOGIN_TOKEN_URL,
+                                                        self._config['auth']['client_secret'],
+                                                        self._config['auth']['client_id'])
+
+        self._next_page_url = self._ONELOGIN_EVENTS_URL
+        return self._get_onelogin_events()
+
+    def _get_onelogin_events(self):
+        """Get all events from the endpoint for this timeframe
+
+        Returns:
+            [
+                {
+                    'id': <int:id>,
+                    'created_at': <str:created_at>,
+                    'account_id': <int:account_id>,
+                    'user_id': <int:user_id>,
+                    'event_type_id': <int:event_type_id>,
+                    'notes': <str:notes>,
+                    'ipaddr': <str:ipaddr>,
+                    'actor_user_id': <int:actor_user_id>,
+                    'assuming_acting_user_id': null,
+                    'role_id': <int:role_id>,
+                    'app_id': <int:app_id>,
+                    'group_id': <int:group_id>,
+                    'otp_device_id': <int:otp_device_id>,
+                    'policy_id': <int:policy_id>,
+                    'actor_system': <str:actor_system>,
+                    'custom_message': <str:custom_message>,
+                    'role_name': <str:role_name>,
+                    'app_name': <str:app_name>,
+                    'group_name': <str:group_name>,
+                    'actor_user_name': <str:actor_user_name>,
+                    'user_name': <str:user_name>,
+                    'policy_name': <str:policy_name>,
+                    'otp_device_name': <str:otp_device_name>,
+                    'operation_name': <str:operation_name>,
+                    'directory_sync_run_id': <int:directory_sync_run_id>,
+                    'directory_id': <int:directory_id>,
+                    'resolution': <str:resolution>,
+                    'client_id': <int:client_id>,
+                    'resource_type_id': <int:resource_type_id>,
+                    'error_description': <str:error_description>
+                }
+            ]
+        """
+        # OneLogin API expects the ISO 8601 format: YYYY-MM-DDTHH:MM:SSZ
+        formatted_date = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%SZ')
+        params = {'since': formatted_date}
```

Comment:
So this is going to actually only get events since the current time. This will likely result in events for only a few seconds or less (the time between setting this date and actually querying their API for the events).

You'll want to use the `self._last_timestamp` for this param, but we'll have to reason about this more since you need a formatted time and not a unix/integer timestamp